### PR TITLE
Create `check-gh-automation` tool

### PR DESCRIPTION
**Pull request by smg247**
A new tool to verify that our bots have access to configured repos.
See the included `README.md` for information about how to use the tool.
There will be a forthcoming presubmit and periodic in `openshift/release` that utilize this.

For: [DPTP-2862](https://issues.redhat.com/browse/DPTP-2862)

### CODE REVIEW

File: cmd/check-gh-automation/main_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	}}
+
+	testCases := []struct {
+		name     string
+		repos    []string
+		bots     []string
+		expected []string
+	}{
+		{
+			name:  "repo has bots as collaborators",
```
Comment (jmguzik) [inline cmd/check-gh-automation/main_test.go:40]: Maybe a case with the error could also be tested in UTs?

**Review by jmguzik — APPROVED** [review]
/lgtm but left a small suggestion.

**Comment by jmguzik** [comment]
@smg247 feel free to remove hold.

File: cmd/check-gh-automation/main.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+		logger.Fatalf("error getting master configs: %v", err)
+	}
+
+	var orgRepos []string
+	if masterConfig.CiOperator != nil && prConfig.CiOperator != nil {
+		changedConfigs, _ := diffs.GetChangedCiopConfigs(masterConfig.CiOperator, prConfig.CiOperator, logger)
+		for _, c := range changedConfigs {
+			orgRepos = append(orgRepos, fmt.Sprintf("%s/%s", c.Info.Org, c.Info.Repo))
+		}
+	}
```
[the remark this answers is not in the document]
Comment (smg247) [reply cmd/check-gh-automation/main.go:152]: That will be much better. I was just utilizing the pattern from `pj-rehearse`. I wasn't aware there was an existing pattern that just used git.

**Comment by smg247** [comment]
While I was testing this out I noticed an instance that it complains about (`stackrox/kernel-packer`) where `openshift-merge-robot` isn't a collaborator, but branch-protection still seems to be enabled and working for that repo. I am going to see if I can figure out a way to tell exactly when that bot must be added.

**Comment by bbguimaraes** [comment]
Very nice.

It looks like `pj-rehearse` takes an amount of time similar to [this](https://github.com/openshift/ci-tools/pull/2834#discussion_r880216421) ([e.g.](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/27092/pull-ci-openshift-release-master-pj-rehearse/1530111269506387968#1:build-log.txt%3A27-28)).  It would be interesting to try to use your new `pkg/config` function there (it would need some additional filtering) and see if it makes a similar difference.  I'm sure people would appreciate rehearsals that start ~30s faster.

/lgtm

**Comment by smg247** [comment]
> It looks like `pj-rehearse` takes an amount of time similar to [this](https://github.com/openshift/ci-tools/pull/2834#discussion_r880216421) ([e.g.](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/27092/pull-ci-openshift-release-master-pj-rehearse/1530111269506387968#1:build-log.txt%3A27-28)). It would be interesting to try to use your new `pkg/config` function there (it would need some additional filtering) and see if it makes a similar difference. I'm sure people would appreciate rehearsals that start ~30s faster.

Good idea! I opened https://issues.redhat.com/browse/DPTP-2888 to track that.

**Pull request merged**